src/components/generator/generatorFunction.js

```diff
@@ -1,131 +1,147 @@
 import arePositionsSimilar from '../../logic/combos/similarPositions';
 
-export function findCombo(vars, retries) {
+/**
+ * Heuristically generates a random combo which adheres to the specified conditions.
+ * @param {*} conditions Conditions that the combo must meet
+ * @param {number} retries Number of times to retry when generating a combo fails
+ */
+export function generateRandomCombo(conditions, retries) {
   const [
-    tricks,
-    positions,
-    numberOfTricks,
-    startFromCheckbox,
-    startFromPosition,
-    allowDuplicates,
-    allowConsecutiveTricks,
-    allowSimilarPositions,
-    allowTransitions,
-    finishToFeet,
-    avgDifficulty,
-    maxDifficulty
-  ] = vars;
-
-  /**
-   * Returns true if the current trick is a legal one in regards to the
-   * specified conditions. This does not check for duplicates as it only
-   * looks at one trick and its predecessor in the combo.
-   */
-  const areLocalComboContitionsLegal = (index, lastTrick, currentTrick) => {
-    const isConsecutiveTricksLegal = allowConsecutiveTricks || lastTrick.id !== currentTrick.id;
-
-    const isEndAndStartEqual = currentTrick.startPos === lastTrick.endPos;
-    const isEndAndStartSimilar = arePositionsSimilar(currentTrick.startPos, lastTrick.endPos);
-    const isTransitionLegal = (allowSimilarPositions && isEndAndStartSimilar)
-                              || isEndAndStartEqual
-                              || allowTransitions;
-
-    if (!isConsecutiveTricksLegal || !isTransitionLegal) {
-      return false;
-    }
+    tricks, positions, numberOfTricks,
+    startFromCheckbox, startFromPosition,
+    allowDuplicates, allowConsecutiveTricks, allowSimilarPositions, allowTransitions, finishToFeet,
+    avgDifficulty, maxDifficulty
+  ] = conditions
+
+  if (!retries) retries = 0
+  if (retries >= 100) {
+    alert("could not find a combo, try changing the settings")
+    return
+  }
 
-    // If the last trick should be to feet, return if that is the case.
-    if (index === numberOfTricks - 1 && finishToFeet) {
-      return ["STAND", "EXPOSURE"].includes(currentTrick.endPos);
-    }
+  if (numberOfTricks <= 1) {
+    alert("You need more than one trick for a combo!")
+    return
+  }
 
-    return true;
+  let randomTricks = new Array(numberOfTricks)
```

Comment:
Is this needed?

Could initializing an empty array `[]`, following by `push`ing onto said array work aswell?

Reply:
Accepted